pkg/provisioner/ironic/ironic.go

```diff
@@ -70,6 +88,29 @@ func init() {
 		fmt.Fprintf(os.Stderr, "Cannot start: No IRONIC_INSPECTOR_ENDPOINT variable set")
 		os.Exit(1)
 	}
+
+	if authStrategy == "http_basic" {
+		ironicUser = os.Getenv("IRONIC_HTTP_BASIC_USERNAME")
+		ironicPassword = os.Getenv("IRONIC_HTTP_BASIC_PASSWORD")
```

Comment:
I wonder if its possible to simplify this.
Do we need to have a non-hard-coded username?
Do we need to have separate passwords for Ironic and Inspector?
Can we just infer the auth strategy from the presence of the data we need?